services/collaboration/pkg/service/grpc/v0/service.go

```diff
@@ -149,8 +152,18 @@ func (s *Service) OpenInApp(
 			return "", err
 		}
 
+		lang := utils.ReadPlainFromOpaque(req.GetOpaque(), "lang")
+
 		q := u.Query()
 		q.Add("WOPISrc", wopiSrcURL.String())
+
+		if s.config.Wopi.DisableChat {
+			q.Add("dchat", "1")
+		}
+
+		q.Add("ui", lang)      // OnlyOffice
+		q.Add("lang", lang)    // Collabora, Impact on the default document language of OnlyOffice
+		q.Add("UI_LLCC", lang) // Office365
```

Comment:
Quoting https://learn.microsoft.com/en-us/microsoft-365/cloud-storage-partner-program/online/discovery#transforming-the-urlsrc-parameter
> Similarly, if the PLACEHOLDER_VALUE is known but the host wishes to ignore it or use the default value for that parameter, you should remove the entire parameter, including the angle brackets.

Sending an empty string isn't a defined behavior, as well as sending additional parameters that aren't defined (for the target WOPI app). I'm fine if we assume that it won't cause problems, but we'll need to monitor this piece of code. In addition, we should probably include a TODO in order to follow the recommended approach at some point.

Reply:
valid point